src/UuidFactory.php

```diff
@@ -178,6 +183,29 @@ public function fromInteger($integer)
         return $this->fromString($hex);
     }
 
+    /**
+     * Checks if a uuid string is valid
+     *
+     * @param  string  $uuid The string to check
+     * @return boolean       Returns true if the string represents a valid UUID; otherwise, returns false
+     */
+    public function isValid($uuid)
+    {
+        // assert or InvalidArgumentException should be used for the integrity of the argument
+        // These statements give helpful indication during debugging
+        assert(func_num_args() === 1);
+        assert('is_string($uuid)');
+
+        $uuid = str_replace(array('urn:', 'uuid:', '{', '}'), '', $uuid);
+
+        // This should probably be moved to the UuidFactory from Uuid
+        if ($uuid === Uuid::NIL || preg_match('/' . self::VALID_PATTERN . '/', $uuid)) {
```

Comment:
@ramsey What are your thoughts on `Uuid::NIL`? Should it be referenced as a constant of the `UuidFactory` as well?